go/vt/vtgate/engine/aggregations.go

```diff
@@ -53,15 +58,26 @@ type AggregateParams struct {
 	CollationEnv *collations.Environment
 }
 
-func NewAggregateParam(opcode opcode.AggregateOpcode, col int, alias string, collationEnv *collations.Environment) *AggregateParams {
+// NewAggregateParam creates a new aggregate param
+func NewAggregateParam(
+	oc opcode.AggregateOpcode,
+	col int,
+	expr evalengine.Expr,
+	alias string,
+	collationEnv *collations.Environment,
+) *AggregateParams {
+	if expr != nil && oc != opcode.AggregateConstant {
+		panic(vterrors.VT13001("expr should be nil"))
```

Comment:
nit: can have better error message, defining why.

Reply:
this is just defensive coding. If this ever happens, we messed up somewhere else. `VT13001` means *bug* - i.e. not expected. not sure which why to write here